Sprint #16

KoRVeRT opened the pull request:
add-docker

KoRVeRT committed:
Sprint #16
add-docker

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/booking/BookingController.java:
```diff
@@ -0,0 +1,85 @@
+                                                 @PathVariable @Positive long bookingId,
+                                                 @RequestParam @NotNull Boolean approved) {
+        log.info("Patch {} for bookingId={} form userId={}",
+                Boolean.TRUE.equals(approved) ? "approval" : "rejection", bookingId, ownerId);
```
Думаю, можно просто использовать `approved`

KoRVeRT replied:
Поменял, но на мой взгляд вкусовшина.

KirPyt commented on gateway/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,92 @@
+    @PostMapping("/{itemId}/comment")
+    public ResponseEntity<Object> postComment(@RequestHeader(USER_ID_HEADER) @Positive Long authorId,
+                                              @PathVariable @Positive long itemId,
+                                              @RequestBody Map<String, String> requestBody) {
```
Здесь стоит создать отдельную ДТО-шку под `Map<String, String> requestBody`, так можно будет автоматически замапить объект и далее с ним удобно работать как с единой структурой

KoRVeRT replied:
ДТО-шка уже была стоило только добавить)

KirPyt commented on gateway/src/test/java/ru/practicum/shareit/booking/BookingControllerTest.java:
```diff
@@ -0,0 +1,107 @@
+    private MockMvc mockMvc;
+
+    @Mock
+    BookingClient bookingClient;
```
Стоит инкапсулировать данное поле

KoRVeRT replied:
Спешил пропустил

KirPyt commented on gateway/src/test/java/ru/practicum/shareit/user/UserControllerTest.java:
```diff
@@ -0,0 +1,132 @@
+    private MockMvc mockMvc;
+
+    @Mock
+    UserClient userClient;
```
Данное поле стоит инкапсулировать)

KoRVeRT replied:
Спешил пропустил

KirPyt commented on server/pom.xml:
```diff
@@ -0,0 +1,102 @@
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
```
Надо убрать валидацию из pom файла сервера

KirPyt commented on server/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -65,8 +60,8 @@ public BookingDto getBookingById(
     public List<BookingDto> getAllBookingByUserId(
             @RequestHeader(USER_ID_HEADER) long userId,
             @RequestParam(required = false, defaultValue = "ALL") String state,
```
При наличии `defaultValue` можно не указывать `required = false`

KoRVeRT replied:
Да заметил перед отправкой, но решил не трогать.

KirPyt commented on server/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -77,8 +72,8 @@ public List<BookingDto> getAllBookingByUserId(
     public List<BookingDto> getAllBookingByOwnerId(
             @RequestHeader(USER_ID_HEADER) long userId,
             @RequestParam(required = false, defaultValue = "ALL") String state,
```
При наличии `defaultValue` можно не указывать `required = false`

KirPyt commented on server/src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -8,12 +8,9 @@
-import javax.validation.constraints.PositiveOrZero;
+
 import java.util.List;
 
```
Надо убрать аннотацию `@Validated` из данного класса

KirPyt commented on server/src/main/java/ru/practicum/shareit/request/controller/ItemRequestController.java:
```diff
@@ -6,9 +6,6 @@
-import javax.validation.constraints.PositiveOrZero;
 import java.util.List;
 
 
```
Надо убрать аннотацию `@Validated` из данного класса

KirPyt commented on server/src/main/java/ru/practicum/shareit/request/controller/ItemRequestController.java:
```diff
@@ -30,7 +27,7 @@ public List<ItemRequestDto> getOwnRequestsByUserId(
+            @RequestBody ItemRequestDto itemRequestDto
     ) {
         itemRequestDto.setRequestorId(userId);
         return itemRequestService.createRequest(itemRequestDto);
```
В каждый метод контроллера стоит добавить логирование входящих запросов

KirPyt commented on server/src/main/java/ru/practicum/shareit/user/controller/UserController.java:
```diff
@@ -33,13 +31,12 @@ public UserDto getUserById(@PathVariable Long userId) {
     }
 
     @PostMapping
```
В каждый метод контроллера стоит добавить логирование входящих запросов

KirPyt commented on server/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -35,7 +30,7 @@ public class BookingController {
-            @Valid @RequestBody BookingDto bookingDto) {
+            @RequestBody BookingDto bookingDto) {
         bookingDto.setBookerId(userId);
         return bookingService.createBooking(bookingDto);
```
В каждый метод контроллера стоит добавить логирование входящих запросов

KirPyt commented on server/src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -44,15 +41,15 @@ public ItemDto getItemById(
     @PostMapping
     public ItemDto createItem(
```
В каждый метод контроллера стоит добавить логирование входящих запросов

KirPyt reviewed (commented):
Приветствую) Я Ваш ревьюер на данную работу. Если возникнут вопросы по ревью, то пишите мне в Пачку

Указал некоторые замечания и советы по коду, предлагаю над ними поработать. Так же некоторые замечания или советы актуальны и для остальных частей кода, поэтому рекомендую обратить внимание и на другие аналогичные места в коде, чтобы всё поправить. Самое важное в данной работе  - это внимательно разделить код между микросервисами gateway и server, убрать валидацию из сервера (валидация в ДТО-шках, контроллерах, pom файле).

KoRVeRT committed:
review #1
corrected minor comments
added a dto for comments
removed validation from the server
added logging to controllers on the server
removed the ability to remove items
added user validation when searching for items by text

KirPyt reviewed (commented):
Посмотрел проект ещё раз - всё очень хорошо) Валидацию с сервера Вы полностью убрали, в остальном рад, что получилось учесть советы по коду

Ваш проект зачтён
